compiler/codegen/codegen.cpp

```diff
@@ -2071,11 +2071,26 @@ void codegen(void) {
   }
 
   // Set the executable name to the name of the file containing the
-  // main module (minus its extension) if it isn't set already.
+  // main module (minus its path and extension) if it isn't set
+  // already.
   if (executableFilename[0] == '\0') {
     ModuleSymbol* mainMod = ModuleSymbol::mainModule();
-    strncpy(executableFilename, mainMod->astloc.filename,
-            sizeof(executableFilename));
+    const char* mainModFilename = mainMod->astloc.filename;
+
+    // find the last slash in the filename's path, if there is one
+    const char* lastSlash = strrchr(mainModFilename, '/');
+    if (lastSlash == NULL) {
+      lastSlash = mainModFilename;
+    } else {
+      lastSlash++;
+    }
+
+    if (strlen(lastSlash) > FILENAME_MAX) {
+      INT_FATAL("input filename exceeds executable filename buffer size");
+    }
+
```

Comment:
Shouldn't the test be ``>=``?

And I'd compare to ``sizeof(executableFilename)`` rather than ``FILENAME_MAX`` in case someone changes the size of the buffer at some point.

Reply:
Oops, so it is.  I'd misremembered confidently enough I didn't look it up again.

Reply:
@dmk42, @cassella: I'm pushing a new commit that compares against sizeof(executableFilename).  It also subtracts one from the strncpy() call to ensure we get a null terminator.  Please double-check my potential off-by-one errors.